documentation/API-design-guidelines.md

```diff
@@ -1116,17 +1101,73 @@ When IpAddr is used in a payload the property objectType MUST be present to indi
 }
 ```
 
-### 11.6 OAuth Definition
+### 11.6 Security definition
+
+This section describes and aligns the security specific definitions within the API specification, such as scopes and the security schemes.
+
+#### 11.6.1 Scope naming
+
+* Define a scope per API operation with the structure:
+
+`api-name:[resource:]action`
+
+where
+
+* `api-name` is the API name specified as the base path, prior to the API version, in the `servers[*].url` property. For example, from `/location-verification/v0`, it would be `location-verification`.
+
+* `resource` is optional. For APIs with several `paths`, it may include the resource in the path. For example, from `/qod/v0/sessions/{sessionId}`, it would be `sessions`.
 
-Finally, this part describes the OAuth security applied to the API. This spec is for client testing purposes only, but
-there should be as similar as possible to the OAuth flows in your production environment. This definition has the
-following aspects:
+* `action`: There are two cases:
+  - For POST operations with a verb, it will be the verb. For example, from `POST /location-verification/v0/verify`, it would be `verify`.
+    - For endpoints designed as POST but with underlying logic retrieving information, a CRUD action `read` may be added, but if it is a path with single operation and it is not expected to have more operations on it, the CRUD action is not necessary.
+  - For CRUD operations on a resource in paths, it will be one of:
+    - `create`: For operations creating the resource, typically `POST`.
+    - `read`: For operations accessing to details of the resource, typically `GET`.
+    - `update`: For operations modifying the resource, typically `PUT` or `PATCH`.
+    - `delete`: For operations removing the resource, typically `DELETE`.
+    - `write` : For operations creating or modifying the resource, when differentiation between `create` and `update` is not needed.
+
+* Eventually we may need to add an additional level to the scope, such as `api-name:[resource:]action[:detail]`, to deal with cases when only a set of parameters/information has to be allowed to be returned. Guidelines should be enhanced when those cases happen.
+
+##### Examples
+
+| API | path | method | scope |
+| --- | --- | --- | --- |
+| location-verification | /verify | POST | `location-verification:verify` |
+| qod | /sessions | POST | `qod:sessions:create`, or<br>`qod:sessions:write` |
+| qod | /qos-profiles | GET | `qod:qos-profiles:read` |
+
+
+#### 11.6.2 Use of openIdConnect for `securitySchemes`
```

Comment:
`openIdConnect` is not the only possibility for security schemes. `clientCredentials` flow is also allowed under certain circumstances. A template for the client credentials flow should also be provided in this document for completeness.

Reply:
@jlurien @eric-murray I understood that this proposal does not prevent to use `clientCredentials`;
With this proposal we extract the authorization information from the yaml. The authorization are described in another json that can be found here: https://example.com/.well-known/openid-configuration (as described here: https://swagger.io/docs/specification/authentication/openid-connect-discovery/)

Did I miss something?

Reply:
So the decision is to use `openIdConnect` as the only possible security scheme?

I guess at least that simplifies the API definition at the cost of not knowing what authentication schemes are supported. The design guidelines should at least contain a note to this effect, because using `openIdConnect` as a way of specifying client credentials is to be used is not particularly intuitive.

Reply:
One more question - would the API definitions that currently call the security scheme of type `openIdConnect` "`three-legged`" be expected to change that to "`openId`"? So rather than have two schemes of type `openIdConnect` called "`three-legged`" and "`two-legged`", we have just one called "`openId`" that covers both cases?

Reply:
This issue was opened prior to the later discussion in I&CM WG about this topic, and the debate is going on there, so in the end we should align Commonalities guidelines with the decision taken there. We have to review how to consolidate relevant information with details in one single place.

Which auth flow is allowed to be used for certain API will depend on many considerations that may change from country to country, that may depend on the business segment, on the considerations about which Personal Information is affected, etc. The idea is of this proposal is to make CAMARA API specifications the more "agnostic" as possible in relation to these concepts.

We are somehow limited here by the semantics in OAS. The ultimate requirement to me is to standardize the scopes which protect the different operations, but for that OAS requires to link the scopes definition in `security` to some entry in the `components/securitySchemes`  section by means of a key. The key itself is free for us to decide. We may choose "openId" or whatever other term, such as something even more generic, such as "scopes", etc. But the entry in securityScheme must be one of the allowed: oauth2, openIdConnect, etc. For sure we may have several entries in security and duplicate the scopes per entry, but using openIdConnect with .well-known discovery URL, which may allow client_credentials if necessary, was to me the most simple option.

Reply:
Yes, I agree that a single `openId` scheme which says "look in `./well-known/openid-configuration` for details" is the most flexible, but is just moving the problem of standardising security schemes across implementations elsewhere.

I would guess we will end up defining a template `openid-configuration` file for each API definition (or multiple files, depending on use case or country) showing what is allowed. Or maybe such details will only be in the documentation.

Reply:
We may agree also in adding some section in info.description for "# Authentication and authorization", explaining how security and securitySchemes should be interpreted.

Reply:
This section is duplicating the work in I&CM, so I will reference the work there